services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -1745,49 +1749,190 @@ class HakemusServiceITest(
+                    val applicationEntity = hakemusRepository.getReferenceById(hakemus.id)
+                    attachmentFactory.save(application = applicationEntity).withContent()
+                    val hakemusData = hakemus.applicationData as KaivuilmoitusData
+                    val founder = hankeKayttajaFactory.getFounderFromHakemus(hakemus.id)
+                    val expectedDataAfterSend =
+                        hakemusData
+                            .copy(pendingOnClient = false)
+                            .setOrdererForContractor(founder.id)
+                    val expectedAlluRequest =
+                        expectedDataAfterSend.toAlluExcavationNotificationData(hakemus.hankeTunnus)
+                    every { alluClient.create(expectedAlluRequest) } returns alluId
+                    justRun { alluClient.addAttachment(alluId, any()) }
+                    justRun { alluClient.addAttachments(alluId, any(), any()) }
+                    every { alluClient.getApplicationInformation(alluId) } returns
+                        AlluFactory.createAlluApplicationResponse(
+                            alluId, applicationId = DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+
+                    val response = hakemusService.sendHakemus(hakemus.id, USERNAME)
+
+                    assertThat(response).all {
+                        prop(Hakemus::alluid).isEqualTo(alluId)
+                        prop(Hakemus::applicationIdentifier)
+                            .isEqualTo(
+                                ApplicationFactory.DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+                        prop(Hakemus::alluStatus).isEqualTo(ApplicationStatus.PENDING)
+                        prop(Hakemus::applicationData).isEqualTo(expectedDataAfterSend)
+                    }
+                    assertThat(hakemusRepository.getReferenceById(hakemus.id)).all {
+                        prop(HakemusEntity::alluid).isEqualTo(alluId)
+                        prop(HakemusEntity::applicationIdentifier)
+                            .isEqualTo(
+                                ApplicationFactory.DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+                        prop(HakemusEntity::alluStatus).isEqualTo(ApplicationStatus.PENDING)
+                    }
+                    verifySequence {
+                        alluClient.create(expectedAlluRequest)
+                        alluClient.addAttachment(alluId, any())
+                        alluClient.addAttachments(alluId, any(), any())
+                        alluClient.getApplicationInformation(alluId)
+                    }
+                }
+            }
 
-            val response = hakemusService.sendHakemus(hakemus.id, USERNAME)
+            @Nested
+            inner class AccompanyingJohtoselvitys {
+
+                private var kaivuilmoitusHakemusId: Long = 0
+                private var cableReportAlluId: Int = 0
+                private var excavationAnnouncenemtAlluId: Int = 0
```

Comment:
The values for these two are always the same, so they can be vals.
```suggestion
                private val cableReportAlluId: Int = alluId
                private val excavationAnnouncenemtAlluId: Int = alluId + 1
```

Reply:
You're right. Fixed them.